crates/engine/tree/src/engine.rs

```diff
@@ -205,6 +206,9 @@ where
                 RequestHandlerEvent::HandlerEvent(HandlerEvent::BackfillAction(action))
             }
             EngineApiEvent::Download(action) => RequestHandlerEvent::Download(action),
+            EngineApiEvent::FatalBlockInsert(error) => {
+                RequestHandlerEvent::HandlerEvent(HandlerEvent::FatalError)
```

Comment:
I'm not sure if this is the right approach here because other places use HandlerEvent::FatalError but they do not pass an error and I believe the context is slightly different so I'm not sure if I should add another handler event or if the existing one should be modified.